src/annotations/modifiers/AnnotationItemEditor.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2020 Damir Porobic <damir.porobic@gmx.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#include "AnnotationItemEditor.h"
+#include "src/annotations/items/interfaces/EditableItem.h"
+
+namespace kImageAnnotator {
+
+AnnotationItemEditor::AnnotationItemEditor()
+	: mCurrentEditItem(nullptr)
+{
+}
+
+void AnnotationItemEditor::handleEditAt(const QPointF &pos, QList<AbstractAnnotationItem *> *items)
+{
+	auto item = findItemAt(pos, items);
+	mCurrentEditItem = dynamic_cast<EditableItem *>(item);
```

Comment:
can we here already notify the item about editing? `mCurrentEditItem->enabledEditing();`
If we take care of it here, we might don't need the stuff in the AnnotationArea and can skip sending the signal.